pytorch_lightning/trainer/lr_finder.py

```diff
@@ -0,0 +1,414 @@
+"""
+Trainer Learning Rate Finder
+"""
+from abc import ABC, abstractmethod
+from typing import Optional
+
+import numpy as np
+import torch
+from torch.optim.lr_scheduler import _LRScheduler
+from torch.utils.data import DataLoader
+from tqdm.auto import tqdm
+import os
+
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning import _logger as log
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class TrainerLRFinderMixin(ABC):
+    @abstractmethod
+    def save_checkpoint(self, *args):
+        """Warning: this is just empty shell for code implemented in other class."""
+
+    @abstractmethod
+    def restore(self, *args):
+        """Warning: this is just empty shell for code implemented in other class."""
+
+    def _run_lr_finder_internally(self, model: LightningModule):
+        """ Call lr finder internally during Trainer.fit() """
+        lr_finder = self.lr_find(model)
+        lr = lr_finder.suggestion()
+        # TODO: log lr.results to self.logger
+        if isinstance(self.auto_lr_find, str):
```

Comment:
Does it also work with something like `hparams.optimizer.lr`? i.e some argument where the LR is not an attribute of hparams but an attribute of an attribute? Can we maybe split keys on every `.`?